src/kakarot/interpreter.cairo

```diff
@@ -895,9 +895,17 @@ namespace Interpreter {
             return (evm, stack, memory, state, 0, 0);
         }
 
-        // Charge the gas fee to the user without setting up a transfer.
-        // Transfers with the exact amounts will be performed post-execution.
-        // Note: balance > effective_fee was verified in AccountContract.execute_from_outside()
+        // // Charge the gas fee to the user without setting up a transfer.
+        // // Transfers with the exact amounts will be performed post-execution.
+        // // Note: balance > effective_fee was verified in AccountContract.execute_from_outside()
+        // let max_gas_fee = gas_limit * tx.max_fee_per_gas;
```

Comment:
@enitrat do you remember if for EIP1559 tx, the balance needs to be greater that the set max_fee_per_gas*gas_limit or only the effective_gas_price*gas_limit?

Reply:
on top of my head, max fee per gas, because it's the maximum possible value that the tx signer is willing to pay. needs a double check

Reply:
I think you are right, in the EIP spec the check is for `max_fee_per_gas`
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md
```
# the signer must be able to afford the transaction
assert signer.balance >= transaction.gas_limit * transaction.max_fee_per_gas
```

Reply:
Won't this be a problem then to do the check in the interpreter ?
The gas_price is set to the effective_gas_price
We do not have access to the transaction.max_fee_per_gas in the interpreter